src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -268,52 +286,288 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
+                    hasErrors = true;
+                }
+                else
+                {
+                    property = LookupPropertyForPattern(inputType, name, out propertyType, ref hasErrors, diagnostics);
+                }
+
+                var boundPattern = BindPattern(pattern, propertyType, hasErrors, diagnostics);
+                builder.Add((property, boundPattern));
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        private Symbol LookupPropertyForPattern(TypeSymbol inputType, IdentifierNameSyntax name, out TypeSymbol propertyType, ref bool hasErrors, DiagnosticBag diagnostics)
+        {
+            var symbol = BindPropertyPatternMember(inputType, name, ref hasErrors, diagnostics);
+
+            if (inputType.IsErrorType() || hasErrors)
+            {
+                propertyType = CreateErrorType();
+                return null;
             }
+
+            propertyType = symbol.GetTypeOrReturnType();
+            return symbol;
         }
 
+        private Symbol BindPropertyPatternMember(
+            TypeSymbol inputType,
+            IdentifierNameSyntax memberName,
+            ref bool hasErrors,
+            DiagnosticBag diagnostics)
+        {
+            // TODO: consider refactoring out common code with BindObjectInitializerMember
+            BoundImplicitReceiver implicitReceiver = new BoundImplicitReceiver(memberName, inputType);
+            string name = memberName.Identifier.ValueText;
+
+            BoundExpression boundMember = BindInstanceMemberAccess(
+                node: memberName,
+                right: memberName,
+                boundLeft: implicitReceiver,
+                rightName: name,
+                rightArity: 0,
+                typeArgumentsSyntax: default(SeparatedSyntaxList<TypeSyntax>),
+                typeArguments: default(ImmutableArray<TypeSymbol>),
+                invoked: false,
+                diagnostics: diagnostics);
+
+            if (boundMember.Kind == BoundKind.PropertyGroup)
+            {
+                boundMember = BindIndexedPropertyAccess(
+                    (BoundPropertyGroup)boundMember, mustHaveAllOptionalParameters: true, diagnostics: diagnostics);
+            }
+
+            LookupResultKind resultKind = boundMember.ResultKind;
+            hasErrors |= boundMember.HasAnyErrors || implicitReceiver.HasAnyErrors;
+
+            switch (boundMember.Kind)
+            {
+                case BoundKind.FieldAccess:
+                case BoundKind.PropertyAccess:
+                    break;
+
+                case BoundKind.IndexerAccess:
+                case BoundKind.DynamicIndexerAccess:
+                case BoundKind.EventAccess:
+                    // PROTOTYPE(patterns2): we need to decide what kinds of members can be used in a property pattern.
+                    // For now we support fields and readable non-indexed properties.
+                default:
+                    if (!hasErrors)
```

Comment:
I can't tell what the desired error approach here is. Is the design: only the first error in a pattern expression is shown? It looks like a lot of error reporting checks this parameter, so I don't know if there's an overarching design goal for preventing cascading errors.

Reply:
It is a goal to suppress cascaded errors, but I would not call it a grand design. It will be adjusted based on practical observations of what works well.

---
In reply to: [154581198](https://github.com/dotnet/roslyn/pull/23209#discussion_r154581198) [](ancestors = 154581198)